arviz/stats/stats.py

```diff
@@ -1341,42 +1363,26 @@ def summary(
     # Combine metrics
     metrics: List[xr.Dataset] = []
     metric_names: List[str] = []
-    if extend:
-        metrics_: Tuple[xr.Dataset, ...]
+    if extend and kind in ["all", "diagnostics"]:
```

Comment:
This block (and the block above where these are computed is hardcoded to the mean stat_focus. The diagnostics shown should also depend on the stat_focus as shown in https://github.com/arviz-devs/arviz/pull/1998/files#r832644368.

The idea I tried to explain in the super long comment should be applied to both stats (more or less done in the provided code example) and diagnostics (no sample code provided)